cert.py

```diff
@@ -144,17 +155,32 @@ def _get_http_challenge(ctx, authzr):
     ctx.fail('no acceptable challenge type found; only HTTP01 supported')
 
 
-def _store_webroot_validation(webroot, challb, val):
+def _store_webroot_validation(ctx, remote, webroot, challb, val):
     logger.info('storing validation of %s' % webroot)
-    try:
-        os.makedirs(os.path.join(webroot, challb.URI_ROOT_PATH), 0o755)
-    except OSError as e:
-        if e.errno != errno.EEXIST:
-            raise
-
-    with open(os.path.join(webroot, challb.path.strip('/')), 'wb') as outf:
-        logger.info('storing validation to %s' % outf.name)
-        outf.write(b(val))
+    chall_path = os.path.join(webroot, challb.path.strip('/'))
+    if not remote:
+        try:
+            os.makedirs(os.path.join(webroot, challb.URI_ROOT_PATH), 0o755)
+        except OSError as e:
+            if e.errno != errno.EEXIST:
+                raise
+
+        with open(chall_path, 'wb') as outf:
+            logger.info('storing validation to %s' % outf.name)
+            outf.write(b(val))
+    else:
+        logger.info('storing validation to %s:%s' % (remote[1], chall_path))
+        ssh = pxssh()
+        ssh.force_password = True
+        ssh.SSH_OPTS = "-o 'NumberOfPasswordPrompts=1'"
+        user, host, port, password = remote
+        try:
+            if ssh.login(host, user, password, port=port):
+                ssh.sendline('echo "%s" > %s' % (b(val), chall_path))
```

Comment:
This makes it hard to have a locked-up account specifically for `wile`. E.g.: you might want to have an account that can only do SFTP inside the `.well-known` folder (i.e.: which doesn't allow arbitrary commands), just in order to allow deploying certificates remotely, without significantly increasing the attack surface of your server.
So maybe using some other way would be more versatile?